repos/re-theme/src/hooks/useStylesMemo.js

```diff
@@ -0,0 +1,28 @@
+/** @module hooks */
+
+import { useMemo } from 'react'
+import { useTheme } from './useTheme'
+import { deepFreeze } from '@keg-hub/jsutils'
+
+const noOpObj = deepFreeze({ content: {} })
+
+/**
+ * Merges styles defined by styleArgs into a memoized style object
+ * @param  {...(string | Object)} styleArgs - any number of strings or style objects. Strings are expected to be theme paths, which will be used to get the associated theme object for that path.
+ * @returns {Object} - result of merging style objects
+ * @example
+ * const textStyles = useStylesMemo('todos.text', textStylesObject)
+ * const myStyles = useStylesMemo(
+ *  'settings.button',
+ *  { main: { margin: 5 }},
+ *  inputStyles,
+ *  'someOtherPath.button'
+ * )
+ */
+export const useStylesMemo = (...styleArgs) => {
```

Comment:
What's the difference between this hook and the `useStyle` hook in keg-components?
Look at => `keg-components/src/hooks/useStyle`

Reply:
I forgot that hook existed. 

This one, `useStylesMemo`, originates from that call we all had a couple of weeks ago  about this topic.

There are two differences I see:
* if `theme.get(...)` returns `undefined`, `useStylesMemo` returns the `noOpObj` that has a `content` key instead. `useStyles` will just return `undefined`. This saves the consumer from needing to null-check the return value.
* `useStyles` recomputes the styles only if one or more of the arguments changed. `useStylesMemo` recomputes it if either the arguments change OR the retheme theme object itself changes. One motivation for this was to recompute when the current window size changes. I don't think `useStyles` would necessarily recompute in that case. Although, with the recent css style changes, that may not be needed?

The only other difference is that it's in `re-theme`, which to me is more appropriate.

What are your thoughts? Now that I see `useStyles` exists, my preference is to delete `useStyles` from re-theme and then rename `useStylesMemo` to `useStyles`.

Reply:
Talked on slack: going to move ahead with deleting `useStyles` from keg-components and then renaming `useStylesMemo` to `useStyles`

Reply:
updated